numpy/core/_add_newdocs.py

```diff
@@ -795,7 +795,10 @@
     ----------
     object : array_like
         An array, any object exposing the array interface, an object whose
-        __array__ method returns an array, or any (nested) sequence.
+        __array__ method returns an array, or any (nested) sequence. However
+        if a number is passed as a parameter then it will return the number itself.
```

Comment:
Also the wording here is a bit confusing - this makes it sound like a pass-through function but in fact a 0-D array is returned (as you note in the next sentence, though the example is incorrect).

My feeling is that the clearest way to add something here would be to note that scalars are array-like and say something like:

> If `object` is a scalar, a 0-dimensional array containing `object` is returned.

Reply:
Hey, @rossbar about the example that I have included here of np.array(2) returning 2, I tried it and I was getting this result so included it here. May you please elaborate more on how this example is wrong?

Reply:
> May you please elaborate more on how this example is wrong?

Sure - in the example you are creating a 0-D array with the value 2 which is distinct from the value that you passed into `np.array`:

```python
>>> v = 2
>>> a = np.array(v)
>>> type(v)
int
>>> type(a)
numpy.ndarray
```

Reply:
A bit of a fly-by comment/thought.  But maybe flipping the logic would be better?  I.e. say: A Python scalar (number), an object convertible to a NumPy array, or any (nested) sequence of these.  <We can continue here explaining what can be converted to NumPy arrays>

Reply:
Thanks, @rossbar & @seberg for your inputs.